affected-paths/core/src/main/kotlin/com/squareup/affected/paths/core/CoreOptions.kt

```diff
@@ -83,5 +87,32 @@ public data class CoreOptions(
     }
   }
 
-  internal val gradleArgs: List<String> = customGradleFlags.toList()
+  internal val gradleArgs: List<String> = buildList {
+    if (autoInjectPlugin) {
+      add("-I")
+      add(
+        File.createTempFile("tempScript", ".gradle").apply {
+          writeText(
+            """
+                allprojects {
+                  buildscript {
+                    repositories {
+                      mavenCentral()
+                    }
+                    dependencies {
+                      classpath "com.squareup.affected.paths:tooling-support:0.1.0"
+                    }
+                  }
+                  
+                  afterEvaluate { p ->
+                    p.apply plugin: "com.squareup.tooling"
+                  }
```

Comment:
nit: use long variable name `project` instead of `p`

Reply:
The reason I used `p` was to prevent confusion as to which `project` variable is being used.

Reply:
My only feedback is that @autonomousapps is likely to discourage the use of allprojects and promote using conventions to apply things.

Reply:
I believe I found an alternative call to use. The scope inside the init script provides a [`Gradle`](https://docs.gradle.org/current/dsl/org.gradle.api.invocation.Gradle.html) instance, which gives a `afterProject {}` closure. This might be the best approach to prevent premature evaluation of the projects.